pontoon/base/views.py

```diff
@@ -396,6 +413,7 @@ def get_translation_history(request):
             "date_iso": t.date.isoformat(),
             "approved_user": User.display_name_or_blank(t.approved_user),
             "unapproved_user": User.display_name_or_blank(t.unapproved_user),
+            "comments": translation_comments_array,
```

Comment:
OK so I'm dumb and I didn't see that @mathjazz had said it all before me. Sorry for making this more confusing! 

You can indeed just do what Matjaz said, and it will work. You should use [`prefetch_related`](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#django.db.models.query.QuerySet.prefetch_related) though to improve query performance, and you *can* set `related_name` to use a different name than `comment_set` but you don't have to.

Reply:
To clarify:  This is the array from `line:396` above which has been serialized and updated with the needed values (at least that is what I thought I did 😀 )

I'm aware of the need to prefetch this data, but am still fuzzy on how to do this and so will be continuing to research on this.

Reply:
Yeah, that array and the entire `for` loop on top of it can go away if you use this: :)
https://github.com/mozilla/pontoon/pull/1524/files/cba6a6eb3f1d7344451d49d1b24173fad1b5cb2a#r360300975.

To prefetch, you  just need to add `.prefetch_related('comments')` to the existing `Translation.objects.filter...` query. To make it work, you need to set the [`related_name`](https://docs.djangoproject.com/en/1.11/ref/models/fields/#django.db.models.ForeignKey.related_name) on the ForeignKey, though.